jlibra-core/src/main/java/dev/jlibra/admissioncontrol/transaction/Program.java

```diff
@@ -1,14 +1,14 @@
 package dev.jlibra.admissioncontrol.transaction;
 
-import java.io.InputStream;
-import java.util.List;
-
+import com.google.protobuf.ByteString;
 import org.immutables.value.Value;
 
+import java.util.List;
+
 @Value.Immutable
 public interface Program {
 
-    InputStream getCode();
+    ByteString getCode();
```

Comment:
If we will get rid of the InputStream, could we instead use some more generic type like byte[] for the data, somehow it just feels wrong having the grpc types used in jlibra classes. It's a bit similar to using java.sql.Dates in application code instead of java generic dates.

Reply:
ByteString is immutable sequence of bytes. If `Program` is to be immutable we can't use mutable byte[]. 
With InputStream the problem is inevitable need of handling IOExceptions at some point which is messy and repetitive. Also about InputStream you never know if someone hasn't already closed the stream or not. Can you use it, or not? ByteString you can get once and feel free to do whatever you want, you will not break stuff.